preprocessing/data_transformer.py

```diff
@@ -213,3 +212,80 @@ def encode_categorical_features(dataframe_dict: dict, columns_list: list,
 
     return dataframe_dict
+
+
+def filter_category(value: str,
+                    to_remove: set) -> str:
+    """ Category filter
+
+    This function resets the value to NAN it it's in corresponding set
+
+    :param str value: a value to be checked.
+    :param set to_remove: a set of values to be reset.
+
+    :return:
+            value: an original value or the NAN-value.
+    """
+    if value in to_remove:
+        return np.nan
+    return value
+
+
+def clean_categorical_features(dataframe_dict: object,
+                               columns_list: object,
+                               inform_only: object = True,
+                               print_results: object = True) -> object:
+    """ Categorical features cleaner
+
+    This function resets the value of a categorical feature to NAN is case it's not exists in both (training and test) datasets.
+    Currently only supports `len(dataframe_dict)=2`: train and test
+
+    :param dict dataframe_dict: a dictionary of Pandas dataframes.
+    :param list columns_list: The list of categorical features to be cleared.
+    :param bool inform_only: If True (default), no data is replaced -  just informing.
+    :param bool print_results: If False, no data is printed to the console. If True, all data is printed to the console.
+
+    :return:
+            dataframe_dict: a dictionary of Pandas dataframes after encoding.
+    """
+
+    # Check if it only one dataframe provided
+    if len(dataframe_dict) != 2:
+        # do nothing and return the original data
+        return dataframe_dict
+
+    # if 2 dataframe than it will be considered as `train` and `test`
+    else:
+        train = dataframe_dict[list(dataframe_dict.keys())[0]]
+        test = dataframe_dict[list(dataframe_dict.keys())[1]]
+
+    # TODO implement support of > 2 dataframes (validation, etc) Here validation set (if any) will be merged to a training data.
+
+    # Checking cycle
+    if print_results:
+        print('*' * 10)
+        print(f"Checking the difference in categorical columns in train and test data: ")
+
+    for column in columns_list:
+        if set(train[column].unique()) != set(test[column].unique()):
+            set_train = set(train[column].unique())
+            set_test = set(test[column].unique())
+            remove_train = set_train - set_test
+            remove_test = set_test - set_train
+            remove = remove_train.union(remove_test)
+
+            # reset values to NAN
+            train[column] = train[column].apply(lambda x: filter_category(x, remove), 1)
+            test[column] = test[column].apply(lambda x: filter_category(x, remove), 1)
+            if print_results:
+                print(f"For column {column} there are {len(remove)} values difference")
+
+    # replace dataframes
+    if not inform_only:
+        dataframe_dict[list(dataframe_dict.keys())[0]] = train
+        dataframe_dict[list(dataframe_dict.keys())[1]] = test
```

Comment:
The same as above

Reply:
Here `train`and `test` are just variable names inside on function. No connection to the real names of dataframes